azkaban-web-server/src/main/bash/azkaban-web-shutdown.sh

```diff
@@ -2,14 +2,33 @@
 # Shutdown script for azkaban web server
 installdir="$(dirname $0)/.."
 
+#currdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+#installdir=$(dirname "${currdir}")
+maxtry=5
 pid=`cat ${installdir}/currentpid`
-echo "Killing Web Server. [pid: $pid]"
 
-kill ${pid}
-if [ $? -ne 0 ]; then
-    echo "Error: Shutdown failed"
-    exit 1;
+if [ -z $pid ]; then
+  echo "currentpid file doesn't exist in ${installdir}, shutdown completed"
+  exit 0
 fi
 
-rm  ${installdir}/currentpid
-echo "done."
+for try in $(seq 1 $maxtry); do
```

Comment:
The logics for the web server and executor are the same. 
Can you think of a way to consolidate the code?